index.js

```diff
@@ -30,9 +30,9 @@ class HLSVideoElement extends CustomVideoElement {
         // autoStartLoad: false,
       });
 
-      hls.loadSource(this.src);
       hls.attachMedia(this.nativeEl);
-    } else if (video.canPlayType('application/vnd.apple.mpegurl')) {
+      hls.on(Hls.Events.MEDIA_ATTACHED, () => hls.loadSource(this.src));
```

Comment:
Hls.js docs seem to recommend [this sequence](https://github.com/video-dev/hls.js#embedding-hlsjs):

```js
hls.loadSource(videoSrc);
hls.attachMedia(video);
```

I haven't looked under the hood at the order of events to see if it makes a difference -- can you perhaps explain the tradeoffs or the difference between the two approaches?

Reply:
It's not clearly explained, but the api doc use this order (the index doc is for quickly play but I guess they made a mistake, I created an issue in hls.js to ask). I think it's to prevent someone to attach something which is not a video ... take a look at https://github.com/video-dev/hls.js/issues/2814

By waiting the video to be attached, you are sure you can then call loadSource.